src/jemalloc.c

```diff
@@ -3716,7 +3721,15 @@ ixallocx_prof(tsd_t *tsd, void *ptr, size_t old_usize, size_t size,
 		prof_info_get(tsd, ptr, alloc_ctx, &prof_info);
 		prof_alloc_rollback(tsd, tctx);
 	} else {
-		prof_info_get_and_reset_recent(tsd, ptr, alloc_ctx, &prof_info);
+		/*
+		 * Need to retrieve the new alloc_ctx since the modification
+		 * to edata has already been done.
```

Comment:
Good catch. Was there a bug before?

Reply:
Yes, the slab used [here](https://github.com/jemalloc/jemalloc/blob/dev/include/jemalloc/internal/arena_inlines_b.h#L96) might be stale if we don't get the new `emap_alloc_ctx`.

Reply:
Maybe separate the bug fix into another PR (to be merged before this one)?

Reply:
Put this change in #2806, also adding a test to cover the `isfree` path above for small & sampled allocs.